src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
the dl library don't exist under OpenBSD. A linkage to -ldl will fail. The dl* function are provide by libc.

BSD system generally don't use libdl for dl function:
DragonFly BSD (and FreeBSD) have the dlopen/dlfcn/... routines located in libc. However many applications and their configure scripts look for these routines in libdl. This adds a stub libdl with no subroutines, so that the configure scripts pass. The dl* routines are automatically available via libc.

Under OpenBSD, this stub don't exist.